### DISCUSSION TITLE
Problems reading output from MF2005 model converted to mf6 despite successful intermediate steps.

### DISCUSSION THREAD

**Discussion by JJHelly** [modflowpy/flopy#1800] [Q&A] [answered]
Aloha.
I am new to modflow and flopy but fairly experience in modeling with quite a few other systems. I suspect I'm stumbling over some well-known conventions that I don't know about yet so pls bear with me.  Here's the problem AFAIK.

1. I've compiled and run mf2005, mf6, and mf6-mpi (including Petsc).
2. I've converted the Maui_fut model that I downloaded from the NSDI links and run it using mf2005.
3. I then converted it with mf5to6 and re-ran it using both mf6 and mf6-mpi (the salinity-related dataset, SWI2, was not converted so don't know the effects of that).
4. I am trying to load it into flopy so I can look at the inputs and outputs.

The latest roadblock is trying to read the sim in this way: 

sim = flopy.mf6.MFSimulation.load('Maui_fut_mf6')

= RESTART: /Volumes/Pegasus32R6/Projects-300/Project-Water-Hawaii/Maui-Water-Balance/data/level1/USGS-Sutra-Data-Maui/Maui_fut_mf6/Maui/mf6/reader.py
loading simulation...
  loading simulation name file...
WARNING: Block "packages" is not a valid block name for file type nam.
  loading tdis package...
Traceback (most recent call last):
  File "/Volumes/Pegasus32R6/Projects-300/Project-Water-Hawaii/Maui-Water-Balance/data/level1/USGS-Sutra-Data-Maui/Maui_fut_mf6/Maui/mf6/reader.py", line 24, in <module>
    sim = flopy.mf6.MFSimulation.load('Maui_fut_mf6')
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/flopy-3.3.7-py3.11.egg/flopy/mf6/modflow/mfsimulation.py", line 777, in load
    for item in models:
TypeError: 'NoneType' object is not iterable

Any suggestions about how to proceed would be quite welcome.

J.

**Comment by christianlangevin** [comment]
I went through the process myself, and this is how it worked for me.  Couple of points:

1. I ran this on a Mac
2. I did not run the mf6 model, though that should not matter
3. You probably need to specify the sim_ws argument to load and give it the path to the folder containing the model
4. FYI, mf6 does not have the SWI package, so the results from mf6 will be quite a bit different

```
(base) mycomputer:input.islandwide.modflow langevin$ mf5to6

 Mf5to6 - Converter for MODFLOW (2005, NWT, LGR) to MODFLOW 6
 Version 6.5.0 Release Candidate 12/09/2022
Usage:  MF5to6  mf2005-name-file  mf6-base-name

Or enter name-file and base-name in response to prompts:

 Enter the name of a MODFLOW-2005, MODFLOW-LGR, or MODFLOW-NWT NAME file:                           
Maui_fut.nam
 Enter the base-name for MODFLOW 6 input:                                                           
mf6


 Converting model based on name file: Maui_fut.nam

 Processing BAS package input...
 Processing LPF package input...
 Processing OC package input...
 Processing GHB package input...
 Processing RCH package input...
 Processing WEL package input...
 Processing DRN package input...

 CONVERSION REPORT
 Original name file: Maui_fut.nam
 New name file: mf6.nam
 -------------------------------------
 Data       New       Volume Budget
 Source     Package   Table Text
 --------   -------   ----------------
 LPF        NPF       [no entry]
 GHB        GHB       HEAD DEP BOUNDS
 RCH        RCH       RECHARGE
 WEL        WEL       WELLS 
 DRN        DRN       DRAINS
 -------------------------------------

NOTES:

 1. In PCG package input, damping and/or relaxation is used. You may need to 
   use under relaxation in IMS input.

WARNINGS:

 Conversion of file type SWI2 is not yet supported.


 Program terminated normally, but see warning(s) above.

(base) mycomputer:input.islandwide.modflow langevin$ ls
Maui_fut.bas			Maui_fut.rch			mf6.drn				mf6.tdis
Maui_fut.dis			Maui_fut.swi2			mf6.ghb				mf6.wel
Maui_fut.drn			Maui_fut.wel			mf6.ic				mf6_conversion_messages.txt
Maui_fut.ghb			RUN_MF2005.BAT			mf6.ims				mfsim.nam
Maui_fut.lpf			fut_rch.dat			mf6.nam				usgs.model.reference
Maui_fut.nam			maui_kani.dat			mf6.npf
Maui_fut.oc			maui_kh.dat			mf6.oc
Maui_fut.pcg			mf6.dis				mf6.rch
(base) mycomputer:input.islandwide.modflow langevin$ ls
Maui_fut.bas			Maui_fut.rch			mf6.drn				mf6.tdis
Maui_fut.dis			Maui_fut.swi2			mf6.ghb				mf6.wel
Maui_fut.drn			Maui_fut.wel			mf6.ic				mf6_conversion_messages.txt
Maui_fut.ghb			RUN_MF2005.BAT			mf6.ims				mfsim.nam
Maui_fut.lpf			fut_rch.dat			mf6.nam				usgs.model.reference
Maui_fut.nam			maui_kani.dat			mf6.npf
Maui_fut.oc			maui_kh.dat			mf6.oc
Maui_fut.pcg			mf6.dis				mf6.rch
(base) mycomputer:input.islandwide.modflow langevin$ conda activate modflow6
(modflow6) mycomputer:input.islandwide.modflow langevin$ ipython
Python 3.10.9 | packaged by conda-forge | (main, Feb  2 2023, 20:26:08) [Clang 14.0.6 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.11.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import flopy

In [2]: sim = flopy.mf6.MFSimulation.load(sim_ws='.')
loading simulation...
  loading simulation name file...
  loading tdis package...
  loading model gwf6...
    loading package dis...
    loading package ic...
WARNING: Block "options" is not a valid block name for file type ic.
    loading package npf...
    loading package oc...
    loading package ghb...
    loading package rch...
    loading package wel...
    loading package drn...
  loading solution package mf6...

In [3]: 
```

I was also able to run this:

```
In [3]: sim.gwf[0].plot()
Out[3]: 
[<Axes: title={'center': ' top'}>,
 <Axes: title={'center': 'botm layer 1'}>,
 <Axes: title={'center': 'idomain layer 1'}>,
 <Axes: title={'center': 'strt layer 1'}>,
 <Axes: title={'center': 'icelltype layer 1'}>,
 <Axes: title={'center': 'k layer 1'}>,
 <Axes: title={'center': ' ghb_0 location stress period 1 layer 1'}>,
 <Axes: title={'center': 'RECHARGE stress period 1'}>,
 <Axes: title={'center': ' wel_0 location stress period 1 layer 1'}>,
 <Axes: title={'center': ' drn_0 location stress period 1 layer 1'}>]

In [4]: import matplotlib

In [5]: import matplotlib.pyplot as plt

In [6]: plt.show()

```

and get this and a few other figures:

![image](https://github.com/modflowpy/flopy/assets/3372403/258790ee-db89-4bf4-8b18-f7f9886b6dba)

**Reply by JJHelly** [reply]
Aloha.  No joy.  Followed your procedure but it doesn't work for me.  Also, your process didn't seem to need a 'mfsim.nam' file whereas mine does; as well as modflowsim.tdis.  Don't know when you ran this but I wonder why the differences.  This doesn't seem to match the documentation for flopy.

Any thoughts?

hellyj@Music-Room mf6 % ls
Maui_fut_mf6.dis			Maui_fut_mf6.ims			Maui_fut_mf6.rch
Maui_fut_mf6.drn			Maui_fut_mf6.nam			Maui_fut_mf6.tdis
Maui_fut_mf6.ghb			Maui_fut_mf6.npf			Maui_fut_mf6.wel
Maui_fut_mf6.ic				Maui_fut_mf6.oc				Maui_fut_mf6_conversion_messages.txt
hellyj@Music-Room mf6 % ln -s Maui_fut_mf6.nam mfsim.nam
hellyj@Music-Room mf6 % python3                         
Python 3.11.3 (v3.11.3:f3909b8bc8, Apr  4 2023, 20:12:10) [Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import flopy
>>> sim = flopy.mf6.MFSimulation.load(sim_ws='.')
loading simulation...
  loading simulation name file...
WARNING: Block "packages" is not a valid block name for file type nam.
  loading tdis package...
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/flopy-3.3.7-py3.11.egg/flopy/mf6/mfpackage.py", line 2628, in load
    fd_input_file = open(
                    ^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Volumes/Pegasus32R6/Projects-300/Project-Water-Hawaii/Maui-Water-Balance/data/level1/USGS-Sutra-Data-Maui/Maui_fut_mf6/mf6/modflowsim.tdis'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/flopy-3.3.7-py3.11.egg/flopy/mf6/modflow/mfsimulation.py", line 758, in load
    instance._tdis_file.load(strict)
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/flopy-3.3.7-py3.11.egg/flopy/mf6/mfpackage.py", line 2637, in load
    raise MFDataException(
flopy.mf6.mfbase.MFDataException: An error occurred in package "None". The error occurred while loading package file in the "load" method.
Additional Information:
(1) File /Volumes/Pegasus32R6/Projects-300/Project-Water-Hawaii/Maui-Water-Balance/data/level1/USGS-Sutra-Data-Maui/Maui_fut_mf6/mf6/modflowsim.tdis of type tdis could not be opened.

**Comment by mbakker7** [comment]
Just a heads up, we are working on a new (and improved) SeaWaterIntrusion (SWI) package for MF6, but this will take probably at least a year to finish. Initial results look great however, so stay tuned.

**Reply by JJHelly** [reply]
Good to know.
Mahalo.

**Comment by JJHelly** [comment]
That's great.  I'll try to reproduce what you did.  Mahalo for responding.

J.
------------------------------------------------------------------------------------------------------------------------------------------------------------
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj / 808 205 9882 / 760 8408660

On May 31, 2023, at 02:37, langevin-usgs ***@***.***> wrote:



I went through the process myself, and this is how it worked for me. Couple of points:

  1.  I ran this on a Mac
  2.  I did not run the mf6 model, though that should not matter
  3.  You probably need to specify the sim_ws argument to load and give it the path to the folder containing the model
  4.  FYI, mf6 does not have the SWI package, so the results from mf6 will be quite a bit different

(base) mycomputer:input.islandwide.modflow langevin$ mf5to6

 Mf5to6 - Converter for MODFLOW (2005, NWT, LGR) to MODFLOW 6
 Version 6.5.0 Release Candidate 12/09/2022
Usage:  MF5to6  mf2005-name-file  mf6-base-name

Or enter name-file and base-name in response to prompts:

 Enter the name of a MODFLOW-2005, MODFLOW-LGR, or MODFLOW-NWT NAME file:
Maui_fut.nam
 Enter the base-name for MODFLOW 6 input:
mf6


 Converting model based on name file: Maui_fut.nam

 Processing BAS package input...
 Processing LPF package input...
 Processing OC package input...
 Processing GHB package input...
 Processing RCH package input...
 Processing WEL package input...
 Processing DRN package input...

 CONVERSION REPORT
 Original name file: Maui_fut.nam
 New name file: mf6.nam
 -------------------------------------
 Data       New       Volume Budget
 Source     Package   Table Text
 --------   -------   ----------------
 LPF        NPF       [no entry]
 GHB        GHB       HEAD DEP BOUNDS
 RCH        RCH       RECHARGE
 WEL        WEL       WELLS
 DRN        DRN       DRAINS
 -------------------------------------

NOTES:

 1. In PCG package input, damping and/or relaxation is used. You may need to
   use under relaxation in IMS input.

WARNINGS:

 Conversion of file type SWI2 is not yet supported.


 Program terminated normally, but see warning(s) above.

(base) mycomputer:input.islandwide.modflow langevin$ ls
Maui_fut.bas                    Maui_fut.rch                    mf6.drn                         mf6.tdis
Maui_fut.dis                    Maui_fut.swi2                   mf6.ghb                         mf6.wel
Maui_fut.drn                    Maui_fut.wel                    mf6.ic                          mf6_conversion_messages.txt
Maui_fut.ghb                    RUN_MF2005.BAT                  mf6.ims                         mfsim.nam
Maui_fut.lpf                    fut_rch.dat                     mf6.nam                         usgs.model.reference
Maui_fut.nam                    maui_kani.dat                   mf6.npf
Maui_fut.oc                     maui_kh.dat                     mf6.oc
Maui_fut.pcg                    mf6.dis                         mf6.rch
(base) mycomputer:input.islandwide.modflow langevin$ ls
Maui_fut.bas                    Maui_fut.rch                    mf6.drn                         mf6.tdis
Maui_fut.dis                    Maui_fut.swi2                   mf6.ghb                         mf6.wel
Maui_fut.drn                    Maui_fut.wel                    mf6.ic                          mf6_conversion_messages.txt
Maui_fut.ghb                    RUN_MF2005.BAT                  mf6.ims                         mfsim.nam
Maui_fut.lpf                    fut_rch.dat                     mf6.nam                         usgs.model.reference
Maui_fut.nam                    maui_kani.dat                   mf6.npf
Maui_fut.oc                     maui_kh.dat                     mf6.oc
Maui_fut.pcg                    mf6.dis                         mf6.rch
(base) mycomputer:input.islandwide.modflow langevin$ conda activate modflow6
(modflow6) mycomputer:input.islandwide.modflow langevin$ ipython
Python 3.10.9 | packaged by conda-forge | (main, Feb  2 2023, 20:26:08) [Clang 14.0.6 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.11.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import flopy

In [2]: sim = flopy.mf6.MFSimulation.load(sim_ws='.')
loading simulation...
  loading simulation name file...
  loading tdis package...
  loading model gwf6...
    loading package dis...
    loading package ic...
WARNING: Block "options" is not a valid block name for file type ic.
    loading package npf...
    loading package oc...
    loading package ghb...
    loading package rch...
    loading package wel...
    loading package drn...
  loading solution package mf6...

In [3]:


I was also able to run this:

In [3]: sim.gwf[0].plot()
Out[3]:
[<Axes: title={'center': ' top'}>,
 <Axes: title={'center': 'botm layer 1'}>,
 <Axes: title={'center': 'idomain layer 1'}>,
 <Axes: title={'center': 'strt layer 1'}>,
 <Axes: title={'center': 'icelltype layer 1'}>,
 <Axes: title={'center': 'k layer 1'}>,
 <Axes: title={'center': ' ghb_0 location stress period 1 layer 1'}>,
 <Axes: title={'center': 'RECHARGE stress period 1'}>,
 <Axes: title={'center': ' wel_0 location stress period 1 layer 1'}>,
 <Axes: title={'center': ' drn_0 location stress period 1 layer 1'}>]

In [4]: import matplotlib

In [5]: import matplotlib.pyplot as plt

In [6]: plt.show()



and get this and a few other figures:

[image]<https://urldefense.com/v3/__https://user-images.githubusercontent.com/3372403/242275658-258790ee-db89-4bf4-8b18-f7f9886b6dba.png__;!!Mih3wA!FXTfNqC3sM4ErHT9isgQCV7suoyT0RZy8UlblR-l7bW-5W7SUaNkpRdgHTZ1aANTofPUuCg6i4vofSuIEnilyI7p$>

**Comment by JJHelly** [comment, marked as answer]
Aloha.

This quandry was solved with the help of Chris Langevin through the flopy Discussion page on github.  
The cause was that I did not copy over the mfsim.nam file after running mf5to6.  Once that was run, I did:

cp Maui_fut_mf6.* ../mf6/.

which only copied the originally-named model files to ./mf6 where I was then running python3 and, of course, that didn't work.
I was under the mis-apprehension that the Maui_fut_mf6.nam file was the only one needed and I was doing the following:

ln -s ./Maui_fut_mf6.nam mfsim.nam

Not realizing that there can be multiple *.nam files and that mf6 uses mfsim.nam as the top-level simulation file.  
I don't know how I would have discovered this otherwise without a complete course in Modflow but I will peruse the manual to see if there is something I could have read to understand this part of the architecture.
